rollouts/api/v1alpha1/rollout_types.go

```diff
@@ -144,9 +144,18 @@ type RolloutStatus struct {
 	// Conditions describes the reconciliation state of the object.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
+	Overall      string       `json:"overall,omitempty"`
+	WaveStatuses []WaveStatus `json:"waveStatuses,omitempty"`
+
 	ClusterStatuses []ClusterStatus `json:"clusterStatuses,omitempty"`
 }
 
+type WaveStatus struct {
+	Name   string `json:"name"`
+	Status string `json:"status"`
+	Paused bool   `json:"paused,omitempty"`
```

Comment:
Should we roll `ClusterStatuses` inside the wave ?

Because to examine progress in a wave, we will need cluster statuses in each of the wave ? right ?

Reply:
I like it. Added cluster statuses to wave status in the latest commit.